### DISCUSSION TITLE
Timeline of bug fixing?

### DISCUSSION THREAD

**Discussion by elephantpanda** [microsoft/onnxruntime-genai#978] [Q&A] [answered]
I noticed there are 58 ongoing "issues". This is more-or-less the same number of open issues that were there 2 months ago.

I just wondered what the timeline is for fixing these bugs. Should I be putting my resources into genai or is this more of an experimental thing that is not expected to be stable for several years?

I appreciate everyone's hard work but if it is still an experimental project with fixing bugs not a high priority it would be great to know in advance so as to focus my resources elsewhere at this time.

Also, I am not entirely clear if genai is open-source or there are full time people working on it, which of course would affect the urgency of fixing the bugs.

Further I am not quite sure who genai is aimed at. If it is for commercial use then having a stable API that does not have any bugs would surely be a high priority?

Or is it a case that the bugs are just too difficult to fix and nobody really knows how to fix them? There is nothing wrong with this, but if that is the case some transparency on this issue would be most refreshing.

**Comment by RyanUnderhill** [comment, marked as answer]
Hello! There's a team working full time on genai, but many of the issues are hard to take action on immediately (model specific issues or feature requests we can't do for a while). It's open source and we welcome outside contributors, but for non trivial changes it's best to discuss them first just to be sure it fits in and will be accepted. I personally am not an expert on the models, just the low level interfacing code.

We're still working on some of the API features (like continuous decoding you commented on previously, it's a big change and it's mostly working).

The library is aimed at anyone wanting to use generative AI at a higher level than what onnxruntime would provide.
